### ISSUE TITLE
Win Folder Redirection causes hangs

### ISSUE THREAD

**Issue by burningTyger** [electron/electron#16531]
* Output of `node_modules/.bin/electron --version`:
v3.1.1
* Operating System (Platform and Version):
Windows 10

**Expected Behavior**
reading path from `app.getPath('documents')` lets me read and write to a user's _Documents_ folder.

**Actual behavior**
Electron doesn't find the folder, can't read from or write to it. The whole process hangs, CPU goes up and I have to force quit.

**To Reproduce**

Use _Folder Redirection, Offline Files, and Roaming User Profiles_ and resirect _Documents_ to a remote server (see official docs here: https://docs.microsoft.com/en-us/windows-server/storage/folder-redirection/folder-redirection-rup-overview)

Basically that means the _Documents_ folder is not in a user's home directory but on a remote share. This somehow confuses electron and I can't read/write to that path.

**Additional Information**
I can't properly reproduce as this is a report from a corporate network. So please close if this doesn't make sense to you. I also can't get logs as I'm not the admin.

**Comment by burningTyger** [comment]
I had the chance to test and change the folders for my setup and noticed that electron sees the redirection as `\\server.domain\foo\bar` instead of `X:\foo\bar` so maybe this is related to the missing drive symbol?
I used another remote share on a different computer (Win 7 though) but this time with a drive symbol and it worked. Sorry for filing such a poor issue but that's all the information I can get my hands on at this time.

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

If this is still reproducible on a supported version, please open a new issue with any other new information that a maintainer should know.

Thank you for taking the time to report this issue and helping to make Electron better! Your help is appreciated.
